src/Modules/OrchardCore.Commerce/Services/ShoppingCartHelpers.cs

```diff
@@ -122,40 +121,22 @@ public class ShoppingCartHelpers : IShoppingCartHelpers
             totals = lines.CalculateTotals().ToList();
         }
 
-        // Checkout should not be possible if any of the items are unpurchasable.
-        var cannotCheckout = false;
         foreach (var line in lines)
         {
             // The values are rounded to avoid storing more precision than what the currency supports.
             line.LinePrice = line.LinePrice.GetRounded();
             line.UnitPrice = line.UnitPrice.GetRounded();
-
-            if (!cannotCheckout)
-            {
-                var productPart = line.Product.ContentItem.As<ProductPart>();
-                if (productPart.As<InventoryPart>() is not { } inventoryPart)
-                {
-                    continue;
-                }
-
-                var item = new ShoppingCartItem(line.Quantity, line.ProductSku, line.Attributes?.Values);
-                var fullSku = _productService.GetOrderFullSku(item, productPart);
-                var inventoryIdentifier = string.IsNullOrEmpty(fullSku) ? productPart.Sku : fullSku;
-                var relevantInventory = inventoryPart.Inventory.FirstOrDefault(entry => entry.Key == inventoryIdentifier);
-
-                cannotCheckout = relevantInventory.Value < 1 &&
-                    !inventoryPart.AllowsBackOrder.Value &&
-                    !inventoryPart.IgnoreInventory.Value;
-            }
         }
 
-        model.IsInvalid = cannotCheckout;
-
+        var model = new ShoppingCartViewModel { Id = cart.Id };
         model.Totals.AddRange(totals.Any() ? totals.Round() : new List<Amount> { new(0, lines[0].LinePrice.Currency) });
-
         model.Headers.AddRange(headers);
         model.Lines.AddRange(lines);
 
+        // Checkout should not be possible if any of the items are unpurchasable.
```

Comment:
Since we no longer set the `model.IsInvalid` on the outside I don't think this comment is needed. This event may be used for other reasons besides validation.